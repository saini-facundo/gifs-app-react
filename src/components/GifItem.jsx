import PropTypes from "prop-types";

function GifItem({ title, url, id }) {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}

export default GifItem;

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
