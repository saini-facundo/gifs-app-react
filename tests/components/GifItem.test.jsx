import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Pruebas en GiftItem", () => {
  const testTitle = "title";
  const testUrl = "https://imgs.com/img.jpg";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={testTitle} url={testUrl} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el URl y el ALT indicado", () => {
    render(<GifItem title={testTitle} url={testUrl} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(testUrl);
    expect(alt).toBe(testTitle);
  });

  test("debe mostar el titulo en el componente", () => {
    render(<GifItem title={testTitle} url={testUrl} />);
    expect(screen.getByText(testTitle)).toBeTruthy();
  });
});
