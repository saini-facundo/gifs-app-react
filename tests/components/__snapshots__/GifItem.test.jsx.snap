// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://imgs.com/img.jpg"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
